### PR TITLE
Governance Page components

### LINKED ISSUE #1851: [TASK] Build up components for future Governance Page
Ideally, it has dropdowns:

- Select the bridge
- Select the contract
- See all governance functions for that contract
- Be able to submit the proposal transactions

Ideally, there's a little component where we list all contracts with respect to this bridge hyperlinked to their addresses on a block explorer. It can change depending on what chain is actively selected in the diagram. It can even show some metadata about the governance settings of the bridge/each side of the bridge, some of the data we have on the stats dApp for example (what tokens are wrappable on the active side, what the fee percentage is)

And to make it even more substantial, we can get more detailed:

For inputs that require a token (addToken, removeToken) we can replace the basic input with a token selector.

### PR DESCRIPTION
**Pull request by vutuanlinh2k2**
## Summary of changes

_Provide a detailed description of proposed changes._

- Implement 2 components for Governance Page: GovernanceContractListingCard and GovernanceContractDetailCard

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [ ] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [x] `libs/webb-ui-components`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. Closes #233)._

- Closes #1851

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/69841784/a5e224be-f660-4e54-a5f3-83516116f41c

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        hideCloseButton
+        {...props}
+        ref={ref}
+      >
+        {/** The search input */}
+        <div className="py-4">
+          <Input
+            id="chain"
+            rightIcon={<Search />}
+            placeholder="Search chains"
```
Comment (drewstone) [inline libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx:44]: I think it's not correct to "Search chains", rather it should be searching contracts. Remember that all the contracts that can be governed exist on all the chains.

Comment (drewstone) [inline libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx:44]:
```suggestion
            placeholder="Search contracts"
```

File: libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      <ListCardWrapper
+        title="Select Contract"
+        hideCloseButton
+        {...props}
+        ref={ref}
+      >
+        {/** The search input */}
+        <div className="py-4">
+          <Input
+            id="chain"
```
Comment (drewstone) [inline libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx:42]: The `id` should also be updated to `contracts` right?

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good besides small nit. The flow is a bit clearer in my head. I think we will
1. Search contracts (they should exist on all chains a bridge exists on)
2. Click contract of interest (paired with the address of the given contract instance)
3. See governance functions
4. Execute governance function -> if chain wasn't selected from the diagram throw a toast error.
5. Otherwise switch chains if needed and execute on the selected chain at that contract address.

**Comment by drewstone** [comment]
This looks good to ship once we address the semantic error between search chains/contracts. If the flow seems correct that's good enough for now @vutuanlinh2k2.

**Comment by vutuanlinh2k2** [comment]
> Looks good besides small nit. The flow is a bit clearer in my head. I think we will
> 
> 1. Search contracts (they should exist on all chains a bridge exists on)
> 2. Click contract of interest (paired with the address of the given contract instance)
> 3. See governance functions
> 4. Execute governance function -> if chain wasn't selected from the diagram throw a toast error.
> 5. Otherwise switch chains if needed and execute on the selected chain at that contract address.

I will move the chain ring to the detail card!

**Pull request merged**